libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/milestone/VoiceInstructionMilestone.java

```diff
@@ -15,21 +16,27 @@
   private DirectionsRoute currentRoute;
   private LegStep currentStep;
   private List<VoiceInstructions> stepVoiceInstructions;
+  private final VoiceInstructionLoader voiceInstructionLoader;
 
   VoiceInstructionMilestone(Builder builder) {
     super(builder);
+    voiceInstructionLoader = VoiceInstructionLoader.getInstance();
   }
 
   @Override
   public boolean isOccurring(RouteProgress previousRouteProgress, RouteProgress routeProgress) {
```

Comment:
👀 https://github.com/mapbox/mapbox-navigation-android/pull/751#discussion_r179104610
We can refactor `isOccurring` deeper.  I see three clear blocks of code here (also you're denoting them adding new lines between them). What about decomposing these conditionals extracting blocks of code into private methods and make intention clearer?

Reply:
Out of scope, that can be another ticket